### ISSUE TITLE
JS codegen. assert statement generates syntactically invalid code.

### ISSUE THREAD

**Issue by yglukhov** [Araq/Nim#2696]
```nim
assert(true)
```
related code in js:
```js
function raiseassert_11879(msg_11881) {
var F={procname:"system.raiseAssert",prev:framePtr,filename:"lib/system.nim",line:0};
framePtr = F;
sysfatal_11884(, msg_11881); // Syntax error here
framePtr = framePtr.prev;
}
```

**Comment by yglukhov** [comment]
Git bisect showed the first bad commit: c5db4fc3a27b17cc20fc13de0a325c126e612c16. @def-, could you take a look please?

**Comment by def-** [comment]
I changed sysFatal from a template to an inline proc. This seems to trigger a bug in the JS backend with procs that take a typedesc: 

```nimrod
type T = object
proc f(t: typedesc, s: string) = discard
f(T, "foobar")
```
This produces wrong JS code for the proc call:
```js
f_14013(, cstrToNimstr("foobar"));
```
Somehow the first argument in the AST ends up being something totally wrong. In the C backend it's just removed, but I'm not sure where that happens and why it doesn't happen in the JS backend.

**Comment by dom96** [comment]
Fixed by #2761
